File: src/main/java/hu/frontrider/arcana/creatureenchant/base/NormalCircle.java
package hu.frontrider.arcana.creatureenchant.base;

import hu.frontrider.arcana.creatureenchant.backend.CreatureEnchant;
import hu.frontrider.arcana.creatureenchant.backend.EnchantingBaseCircle;
import net.minecraft.entity.EntityLivingBase;
import thaumcraft.api.aspects.Aspect;
import thaumcraft.api.aspects.AspectList;

import static hu.frontrider.arcana.ThaumicArcana.MODID;

public class NormalCircle extends EnchantingBaseCircle {

    private Color color;
    public NormalCircle(){
        setRegistryName(MODID,"normal");
        color = new Color(255,255,255,255);
    }

    @Override
    public Color getColor() {
        return color;
    }

    @Override
    public int doEffect(int multiplier, EntityLivingBase entityLiving, CreatureEnchant enchant) {
        return multiplier;
    }

    @Override
    public AspectList getFormula() {
        return new AspectList()
                .merge(Aspect.BEAST,300)
                .merge(Aspect.MAGIC,100)
                .merge(Aspect.LIFE,50)
                .merge(Aspect.AURA,100);
    }
}

File: src/main/java/hu/frontrider/arcana/creatureenchant/base/NormalCircle.kt
package hu.frontrider.arcana.creatureenchant.base

import net.minecraft.entity.EntityLivingBase
import thaumcraft.api.aspects.Aspect
import thaumcraft.api.aspects.AspectList

import hu.frontrider.arcana.ThaumicArcana.MODID
import hu.frontrider.arcana.creatureenchant.CreatureEnchant
import hu.frontrider.arcana.creatureenchant.EnchantingBaseCircle

class NormalCircle : EnchantingBaseCircle("normal") {

    override val formula = AspectList()
            .merge(Aspect.BEAST, 300)
            .merge(Aspect.MAGIC, 100)
            .merge(Aspect.LIFE, 50)
            .merge(Aspect.AURA, 100)

    override val color= Color(255, 255, 255, 255)

    override val research = "ENCHANT_MODIFICATION"

    init {
        setRegistryName(MODID, "normal")

    }

    override fun doEffect(multiplier: Int, entityLiving: EntityLivingBase, enchant: CreatureEnchant): Int {
        return multiplier
    }
}
